index.js

```diff
@@ -68,9 +68,48 @@ async function run() {
           containerDef.environment.push(variable);
         }
       })
+    if (secrets) {
+        // If environment array is missing, create it
+        if (!Array.isArray(containerDef.secrets)) {
+          containerDef.secrets = [];
+        }
+
+        // Get pairs by splitting on newlines
+        secrets.split('\n').forEach(function (line) {
+          // Trim whitespace
+          const trimmedLine = line.trim();
+          // Skip if empty
+          if (trimmedLine.length === 0) { return; }
+          // Split on =
+          const separatorIdx = trimmedLine.indexOf("=");
+          // If there's nowhere to split
+          if (separatorIdx === -1) {
+              throw new Error(
+                `Cannot parse the secret '${trimmedLine}'. Secret pairs must be of the form NAME=valueFrom, 
+                where valueFrom is an arn from parameter store or secrets manager. See AWS documentation for more information: 
+                https://docs.aws.amazon.com/AmazonECS/latest/developerguide/specifying-sensitive-data.html.`);
+          }
```

Comment:
Ideally, this (Lines 80 to Lines 91, I think) could be transferred to a common method as we are using same set of code when fetching `environmentVariables` as well.